### PR TITLE
feat: add MeshGateway support to MeshAccessLog

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist prior to review

`> Changelog: skip`

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? -- #4979 
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests -- none
- [x] Manual Kubernetes Tests -- none
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/plugin.go [hunk 1]
```diff
@@ -145,9 +158,50 @@ func applyToDirectAccess(
...
+	if rawList := resources[core_mesh.MeshGatewayType]; rawList != nil {
+		gateways = rawList.(*core_mesh.MeshGatewayResourceList)
+	} else {
+		return nil
+	}
+
+	gateway := xds_topology.SelectGateway(gateways.Items, dataplane.Spec.Matches)
+	if gateway == nil {
+		return nil
+	}
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/plugin.go:174]: what do you think about enriching `Proxy` object and moving this logic to `DataplaneProxyBuilder`?

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/plugin.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/plugin.go:174]: Definitely, fixing `Proxy` has been on my list for a while

Comment (michaelbeaumont) [reply pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/plugin.go:166]: https://github.com/kumahq/kuma/issues/5124

**Comment by michaelbeaumont** [comment]
@jakubdyszkiewicz gonna merge this, let me know if there's anything missing in the e2e test.

**Pull request merged**